File: BlogModule.Infrastructure/Mappers/BlogPostCommentPersistenceMapper.cs
using BlogModule.Domain.Models;
using BlogModule.Infrastructure.DataBase.Entities;
using Shared.Utilities;

namespace BlogModule.Infrastructure.Mappers
{
    public class BlogPostCommentPersistenceMapper : IPersistenceMapper<BlogPostCommentEntity, BlogPostCommentModel>
    {
        public BlogPostCommentModel MapToModel(BlogPostCommentEntity entity) => new()
        {
            Id = entity.Id,
            Description = entity.Description,
            BlogPostId = entity.BlogPostId,
            UserId = entity.UserId,
            DateAdded = entity.DateAdded
        };

        public BlogPostCommentEntity MapToEntity(BlogPostCommentModel model) => new()
        {
            Id = model.Id,
            Description = model.Description,
            BlogPostId = model.BlogPostId,
            UserId = model.UserId,
            DateAdded = model.DateAdded
        };
    }
}

File: BlogApi/Infrastructure/databases/postgres/mappers/blog_post_comment_persistence_mapper.go
package mappers

import (
	models "identity-api/Domain/models/domestic"
	"identity-api/Infrastructure/databases/postgres/entities"
	"time"

	"github.com/google/uuid"
)

type BlogPostCommentMapper struct{}

func NewBlogPostCommentPersistenceMapper() *BlogPostCommentMapper {
	return &BlogPostCommentMapper{}
}

func (m *BlogPostCommentMapper) MapToModel(entity entities.BlogPostCommentEntity) models.BlogPostCommentModel {
	return models.BlogPostCommentModel{
		Id:          entity.ID.String(),
		BlogPostId:  entity.BlogPostID.String(),
		UserId:      entity.UserID.String(),
		Description: entity.Description,
		CreatedAt:   entity.CreatedAt.Format(time.RFC3339),
		UpdatedAt:   entity.UpdatedAt.Format(time.RFC3339),
	}
}

func (m *BlogPostCommentMapper) MapToEntity(model models.BlogPostCommentModel) entities.BlogPostCommentEntity {
	var id uuid.UUID
	if model.Id == "" {
		id = uuid.New()
	} else {
		id, _ = uuid.Parse(model.Id)
	}

	blogPostID, _ := uuid.Parse(model.BlogPostId)
	userID, _ := uuid.Parse(model.UserId)

	return entities.BlogPostCommentEntity{
		ID:          id,
		BlogPostID:  blogPostID,
		UserID:      userID,
		Description: model.Description,
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
}
